File: neetcode/128. Longest Consecutive Sequence/128.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length <= 1) return nums.length;

        Arrays.sort(nums);

        int currLen = 1;
        int maxLen = 1;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] - nums[i - 1] == 0) continue;
            if (nums[i] - nums[i - 1] == 1) {
                currLen++;
                maxLen = Math.max(maxLen, currLen);
            } else {
                currLen = 1;
            }
        }

        return maxLen;
    }
}

File: neetcode/128. Longest Consecutive Sequence/128.kt
class Solution {
    fun longestConsecutive(nums: IntArray): Int {
        val n = nums.size
        if (n <= 1) return n

        nums.sort()
        var currLen = 1
        var maxLen = 1

        for (i in 1 until n) {
            var diff = nums[i] - nums[i - 1]
            if (diff == 0) continue

            if (diff == 1) {
                currLen++
                maxLen = max(maxLen, currLen);
            } else {
                currLen = 1
            }
        }

        return maxLen;
    }
}
